### PR TITLE
Test

### PR DESCRIPTION
**Pull request by superjax**
Unit tests for math library, state machine, rc override and command manager

### CODE REVIEW

**Comment by superjax** [comment]
Closes #195

File: test/command_manager_test.cpp [hunk 1]
```diff
@@ -0,0 +1,610 @@
...
+  //=================================================
+
+  // Only override attitude
+  rf.params_.set_param_int(PARAM_RC_ATTITUDE_OVERRIDE_CHANNEL, 4);
+  rf.params_.set_param_int(PARAM_RC_THROTTLE_OVERRIDE_CHANNEL, 4);
+
+  // switch is off, but roll channel is deviated
+  rc_values[4] = 1100;
+  rc_values[0] = 1800;
+  rc_values[2] = 1000;
```
Comment (dpkoch) [inline test/command_manager_test.cpp]: What is `rc_values[2]` doing?

File: test/command_manager_test.cpp [hunk 2]
```diff
@@ -0,0 +1,610 @@
...
+  rf.command_manager_.set_new_offboard_command(offboard_command);
+  step_firmware(rf, board, 20000);
+  output = rf.command_manager_.combined_control();
+  EXPECT_PRETTYCLOSE(output.F.value, 0.2);
+
+
+
+  // Okay, remove the Min throttle setting
+  rf.params_.set_param_int(PARAM_RC_OVERRIDE_TAKE_MIN_THROTTLE, false);
+
```
Comment (dpkoch) [inline test/command_manager_test.cpp]: Let's add a test here to make sure that min throttle is actually off now?

File: test/command_manager_test.cpp [hunk 3]
```diff
@@ -0,0 +1,610 @@
...
+  float max_roll = rf.params_.get_param_float(PARAM_RC_MAX_ROLL);
+  float max_pitch = rf.params_.get_param_float(PARAM_RC_MAX_PITCH);
+  float max_yawrate = rf.params_.get_param_float(PARAM_RC_MAX_YAWRATE);
+
+  // Let's clear all errors in the state_manager
+  rf.state_manager_.clear_error(rf.state_manager_.state().error_codes);
+
+  //=================================================
+  // RC Arming Test
+  //=================================================
```
Comment (dpkoch) [inline test/command_manager_test.cpp]: I wonder if these arming tests should get moved into the state manager tests?

File: test/command_manager_test.cpp [hunk 4]
```diff
@@ -0,0 +1,610 @@
...
+  EXPECT_EQ(rf.state_manager_.state().failsafe, false);
+
+  // flip the arm switch off
+  rc_values[4] = 1100;
+  board.set_rc(rc_values);
+  step_firmware(rf, board, 20000);
+  // we should be disarmed
+  EXPECT_EQ(rf.state_manager_.state().armed, false);
+  EXPECT_EQ(rf.state_manager_.state().error, false);
+  EXPECT_EQ(rf.state_manager_.state().failsafe, false);
```
Comment (dpkoch) [inline test/command_manager_test.cpp]: Let's also add the following tests:
  - sticks take the appropriate amount of time to arm
  - sticks don't arm when they shouldn't
  - switch doesn't arm when it shouldn't (maybe just set all of the switch channels high to cover them all and make sure it doesn't arm)

File: test/command_manager_test.cpp (hunk 3 above)
Comment (superjax) [inline test/command_manager_test.cpp]: We could do that.  I figured that they were extensive enough that they justified their own file.  They used to be a bunch of separate executables, but I combined them to make it easier to run the test.

Comment (superjax) [inline test/command_manager_test.cpp]: I'm mostly concerned with how tong the ifles get

File: test/command_manager_test.cpp (hunk 4 above)
Comment (superjax) [inline test/command_manager_test.cpp]: cool

File: test/command_manager_test.cpp [hunk 5]
```diff
@@ -0,0 +1,610 @@
...
+  while (board.clock_micros() < 1000000)
+  {
+    step_firmware(rf, board, 20000);
+  }
+
+  offboard_command.stamp_ms = board.clock_millis();
+  rf.command_manager_.set_new_offboard_command(offboard_command);
+  step_firmware(rf, board, 20000);
+
+  // We don't have an override channel mapped, so this should be RC
```
Comment (superjax) [inline test/command_manager_test.cpp]: yup

File: test/command_manager_test.cpp (hunk 1 above)
Comment (superjax) [inline test/command_manager_test.cpp]: moving the throttle to neutral position

**Comment by dpkoch** [comment]
Our Travis builds should run a little faster now; turns out it by default assumes you're using ruby and downloads and installs a bunch of ruby stuff. By specifying C++ as the language it skips all of that.

**Comment by superjax** [comment]
Cool!  Thanks

**Pull request merged**